lib/Migrator/APIDiffMigratorPass.cpp

```diff
@@ -1290,6 +1292,94 @@ struct APIDiffMigratorPass : public ASTMigratorPass, public SourceEntityWalker {
+    llvm::StringSet<> Results;
+    Results.insert("application(_,delegateHandlesKey)");
+    Results.insert("changeColor(_)");
+    Results.insert("controlTextDidBeginEditing(_)");
+    Results.insert("controlTextDidEndEditing(_)");
+    Results.insert("controlTextDidChange(_)");
+    Results.insert("changeFont(_)");
+    Results.insert("validModesForFontPanel(_)");
+    Results.insert("discardEditing()");
+    Results.insert("commitEditing()");
+    Results.insert("commitEditing(withDelegate,didCommit,contextInfo)");
+    Results.insert("commitEditingAndReturnError()");
+    Results.insert("objectDidBeginEditing(_)");
+    Results.insert("objectDidEndEditing(_)");
+    Results.insert("validateMenuItem(_)");
+    Results.insert("pasteboard(_,provideDataForType)");
+    Results.insert("pasteboardChangedOwner(_)");
+    Results.insert("validateToolbarItem(_)");
+    Results.insert("layer(_,shouldInheritContentsScale,from)");
+    Results.insert("view(_,stringForToolTip,point,userData)");
+    return Results;
+  }
+
+  SourceLoc shouldRemoveOverride(AbstractFunctionDecl *AFD) {
+    if (AFD->getKind() != DeclKind::Func)
+      return SourceLoc();
+    SourceLoc OverrideLoc;
+
+    // Get the location of override keyword.
+    if (auto *Override = AFD->getAttrs().getAttribute<OverrideAttr>()) {
+      if (Override->getRange().isValid()) {
+        OverrideLoc = Override->getLocation();
+      }
+    }
+    if (OverrideLoc.isInvalid())
+      return SourceLoc();
+    SmallVector<ParamDecl*, 4> Params;
+    collectParamters(AFD, Params);
+    llvm::SmallString<32> Buffer;
+    llvm::raw_svector_ostream OS(Buffer);
+    OS << AFD->getNameStr();
+    OS << "(";
+    for (auto *P: Params) {
+      auto Arg = P->getArgumentName();
+      if (P != Params.front())
+        OS << ",";
+      if (Arg.empty()) {
+        OS << "_";
+      } else {
+        OS << Arg.str();
+      }
+    }
+    OS << ")";
+    return OverridingRemoveNames.find(OS.str()) == OverridingRemoveNames.end() ?
+      SourceLoc() : OverrideLoc;
+  }
+
+  struct SuperRemoval: public ASTWalker {
+    EditorAdapter &Editor;
+    SuperRemoval(EditorAdapter &Editor): Editor(Editor) {}
+    static bool isSuperExpr(Expr *E) {
+
+      // Check if the expression is super.foo().
+      if (auto *CE = dyn_cast<CallExpr>(E)) {
+        if (auto *DSC = dyn_cast<DotSyntaxCallExpr>(CE->getFn())) {
+          return DSC->getBase()->getKind() == ExprKind::SuperRef;
+        }
+      }
+      // We should handle try super.foo() too.
+      if (auto *TE = dyn_cast<AnyTryExpr>(E)) {
+        return isSuperExpr(TE->getSubExpr());
+      }
+      return false;
+    }
+    std::pair<bool, Expr*> walkToExprPre(Expr *E) override {
+      // We should only handle top level expressions, so avoid visiting
+      // sub-expressions.
+      auto Result = std::make_pair(false, E);
+      if (E->isImplicit())
+        return Result;
+      if (isSuperExpr(E)) {
+        Editor.remove(E->getSourceRange());
```

Comment:
Shouldn't this check if the thing it's referencing is one of the targeted methods? E.g. I might be calling some unrelated method on NSObject that I also happen to override and I want to call the super class' version rather than my own.

Reply:
Good point! I'll add this check for these function invocation.